src/Compilers/CSharp/Portable/Symbols/MemberSymbolExtensions.cs

```diff
@@ -517,5 +517,13 @@ internal static string GetMemberCallerName(this Symbol member)
                 member.IsExplicitInterfaceImplementation() ? ExplicitInterfaceHelpers.GetMemberNameWithoutInterfaceName(member.Name) :
                 member.Name;
         }
+
+        internal static void EnsureNullableAttributeExistsIfNecessary(this Symbol symbol, TypeSymbolWithAnnotations type, DiagnosticBag diagnostics)
+        {
+            if (type.ContainsNullableReferenceTypes())
+            {
+                symbol.DeclaringCompilation.EnsureNullableAttributeExists(diagnostics, symbol.Locations[0], modifyCompilationForNullable: true);
```

Comment:
Are we sure `symbol.Location.Length` is always > 0?

Reply:
No, `Locations[0]` was not correct. Fixed.